lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,29 +1,921 @@
+        self.camera = HDU[1].header['CAMERA']
+        self.ccd = HDU[1].header['CCD']
+
+        # Get the requested cbv_type
+        switcher = {
+            'SingleScale': 'CBV.single-scale.{}.{}'.format(self.camera, self.ccd),
+            'MultiScale': 'CBV.multiscale-band-{}.{}.{}'.format(self.band,
+                self.camera, self.ccd),
+            'Spike': 'CBV.spike.{}.{}'.format(self.camera, self.ccd),
+            'unknown': 'error'
+            }
+        extName = switcher.get(self.cbv_type, switcher['unknown'])
+        if (extName == 'error'):
+            raise Exception('Invalide cbv_type')
+
+
+        cbv_data = HDU[extName].data
+
+        self.cadenceno      = HDU[extName].data['CADENCENO']
+        self.gap_indicators = HDU[extName].data['GAP']
+        self.cbvEXTNAME     = HDU[extName].header['EXTNAME']
+
+        # Pull out each individual CBV
+        self.cbv_array, self.cbv_indices = self._extract_cbvs_from_hdu_data(cbv_data, self.cbv_indices)
+
+    @staticmethod
+    def get_tess_cbv_url(sector, camera, CCD):
+        """ STATIC method to obtain a path to a TESS CBV FITS file
+
+            The easiest way to obtain a link to the CBV file for a TESS Sector and camera.CCD is
+
+            1. Download the bulk download curl script (with a predictable url) for the desired sector and search it for the camera.CCD I need
+            2. Download the CBV FITS file based on the link in the curl script
+
+            The bulk download curl links have urls such as:
+
+            https://archive.stsci.edu/missions/tess/download_scripts/sector/tesscurl_sector_17_cbv.sh
+
+            Then the individual CBV files foudn in the curl file have urls such as:
+
+            https://archive.stsci.edu/missions/tess/ffi/s0017/2019/279/1-1/tess2019279210107-s0017-1-1-0161-s_cbv.fits
+
+            Returns a url string for the desired CBV FITS file
+    
+        """
+        curlBaseUrl = 'https://archive.stsci.edu/missions/tess/download_scripts/sector/tesscurl_sector_'
+        curlEndUrl = '_cbv.sh'
+        curlUrl = curlBaseUrl + str(sector) + curlEndUrl
+
+        # This is the string to search for in the curl script file
+        # Pad the sector number with a first '0' if less than 10
+        # TODO: figure out a way to pad an interger number with forward zeros
+        # without needing conditional
+        sector = int(sector)
+        if (sector < 10):
+            curlSearchString = 's000' + str(sector) + '-' + str(camera) + '-' + str(CCD) + '-'
+        elif (sector > 99):
+            # We will be blessed if we get to more than 99 sectors!
+            raise Exception('Only up to 99 Sectors is currently supported')
+        else:
+            curlSearchString = 's00' + str(sector) + '-' + str(camera) + '-' + str(CCD) + '-'
+
+        # 1. Read in the relevent curl script file and find the line for the CBV data we are looking for
+        data = urllib.request.urlopen(curlUrl)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.](https://app.codacy.com/manual/barentsen/lightkurve/pullRequest?prid=4896575)